lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -633,14 +642,24 @@ def raw_inhibit_warnings_hash
       # Returns the inhibit_warnings hash pre-populated with default values.
       #
       # @return [Hash<String, Array>] Hash with :all key for inhibiting all
-      #         warnings, and :for_pods key for inhibiting warnings per Pod.
+      #         warnings, :for_pods key for inhibiting warnings per Pod,
+      #         and :not_for_pods key for not inhibiting warnings per Pod.
       #
       def inhibit_warnings_hash
         inhibit_hash = raw_inhibit_warnings_hash
         if exclusive?
           inhibit_hash
         else
-          parent.send(:inhibit_warnings_hash).merge(inhibit_hash) { |l, r| (l + r).uniq }
+          parent_hash = parent.send(:inhibit_warnings_hash)
+          if parent_hash['not_for_pods']
+            parent_hash['not_for_pods'] -= inhibit_hash['for_pods']
```

Comment:
Hmm yeah, I think `parent_hash.dup` would be enough, do you see any other possible issues on that?